sdk/template/template/src/tracing.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getTracer } from "@azure/core-tracing";
+import { Span, SpanOptions, SpanKind } from "@opentelemetry/api";
+import { OperationOptions } from "@azure/core-http";
+
+type OperationTracingOptions = OperationOptions["tracingOptions"];
+
+/**
+ * Creates a span using the global tracer.
+ *
+ * @param name - The name of the operation being performed.
+ * @param tracingOptions - The options for the underlying http request.
```

Comment:
tag it with `@internal`? Also make sure to use `@internal` for all such definitions not exported by `src/index.ts`.